impl/src/main/java/jakarta/faces/event/PhaseId.java

```diff
@@ -16,19 +16,24 @@
 
 package jakarta.faces.event;
 
-import java.util.Arrays;
-import java.util.Collections;
+import jakarta.faces.FacesException;
+
 import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Stream;
 
-import jakarta.faces.FacesException;
+import static java.util.Collections.unmodifiableMap;
+import static java.util.function.Function.identity;
+import static java.util.stream.Collectors.toMap;
 
 /**
  * <p>
  * <span class="changed_modified_2_2">Typesafe</span> enumeration of the legal values that may be returned by the
  * <code>getPhaseId()</code> method of the {@link FacesEvent} interface.
  */
 
-public class PhaseId implements Comparable {
+public class PhaseId implements Comparable<PhaseId> {
```

Comment:
This is a public API change. You can't do that without a ticket in https://github.com/jakartaee/faces/issues and referencing the PR to it.